### PR TITLE
SDI-367: 🔒️ Add security to move to hospital endpoints

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: integration_tests/.eslintrc [hunk 1]
```diff
@@ -1,10 +1,11 @@
 {
-  "plugins": ["cypress"],
+  "plugins": ["cypress", "no-only-tests"],
```
Comment (petergphillips) [inline integration_tests/.eslintrc:2]: realised that no only was missing from this project 😠

File: integration_tests/integration/movePrisoner.cy.js [hunk 2]
```diff
@@ -17,13 +17,9 @@ const toOffender = $cell => ({
 context('Move prisoner', () => {
   beforeEach(() => {
     cy.task('reset')
-    cy.task('stubLogin')
+    cy.task('stubLogin', { roles: ['TRANSFER_RESTRICTED_PATIENT'] })
     cy.task('stubAuthUser')
-    cy.task('stubUserRoles', [
-      { roleCode: 'REMOVE_RESTRICTED_PATIENT' },
```
Comment (petergphillips) [inline integration_tests/integration/movePrisoner.cy.js:23]: only needs the transfer role

File: server/middleware/authorisationMiddleware.ts [hunk 3]
```diff
@@ -4,14 +4,17 @@ import { RequestHandler } from 'express'
 import logger from '../../logger'
 import asyncMiddleware from './asyncMiddleware'
 
-export default function authorisationMiddleware(authorisedRoles: string[] = []): RequestHandler {
+export default function authorisationMiddleware(
+  renderNotFound: boolean,
```
Comment (petergphillips) [inline server/middleware/authorisationMiddleware.ts:8]: we want a not found page in move prisoner and other routes

File: server/routes/movePrisoner/index.ts [hunk 4]
```diff
@@ -18,6 +19,7 @@ export default function movePrisonerRoutes({
   prisonerSearchService: PrisonerSearchService
 }): Router {
   const router = express.Router({ mergeParams: true })
+  router.use(authorisationMiddleware(true, ['TRANSFER_RESTRICTED_PATIENT']))
```
Comment (petergphillips) [inline server/routes/movePrisoner/index.ts:22]: restrict all routes under move prisoner endpoints

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**